.ci/bump-license-year.sh

```diff
@@ -9,7 +9,7 @@ DIR=$3
 OLD_VALUE="// Copyright (C) 2001-$PREV_YEAR the original author or authors."
 NEW_VALUE="// Copyright (C) 2001-$CURR_YEAR the original author or authors."
 
-find "$DIR" -type f \( -name *.java -o -name *.header -o -name *.g4 \) \
+find "$DIR" -type f \( -name '*.java' -o -name '*.header' -o -name '*.g4' \) \
```

Comment:
@shubh220922 please run new find command and old find command using subshsell redirection and `diff` command, like `diff <(find ...) <(find..)` and share output here.

Reply:
You can replace `"$DIR"` with `.`, ran from root of project

Reply:
@nrmancuso 

I tried running the command you have shared multiple times (one such example is below) but, I am not able to properly diff the results of the command
```shell
╭─shubh220922@fedora ~/Desktop/checkstyle_project/checkstyle ‹sc2061●› 
╰─$ diff <(find . -type f ( -name *.java -o -name *.header -o -name *.g4) -exec sed -i "s|$OLD_VALUE|$NEW_VALUE|g") <(find . -type f ( -name '*.java' -o -name '*.header' -o -name '*.g4') -exec sed -i "s|$OLD_VALUE|$NEW_VALUE|g")
zsh: unknown file attribute:  
zsh: unknown file attribute:  
╭─shubh220922@fedora ~/Desktop/checkstyle_project/checkstyle ‹sc2061●› 
╰─$ 
```

Reply:
@shubh220922 it looks like you are using zsh instead of bash. I recommend that you stick with the interpreter defined in the script that you are running commands from, especially when dealing with wildcards. bash automatically expands wildcards, but zsh does not.

Reply:
@nrmancuso I ran it in bash and it worked. But please cross verify is the command I ran is correct or not.

```shell
[shubh220922@fedora checkstyle]$ diff <(find . -type f \( -name *.java -o -name *.header -o -name *.g4 \)) <(find . -type f \( -name '*.java' -o -name '*.header' -o -name '*.g4' \))
[shubh220922@fedora checkstyle]$
``` 

No output was found. Problem happened in zsh for some reason.

Reply:
>please cross verify if the command I ran is correct or not.

command looks good :)

>Problem happened in zsh for some reason.

No need to wonder, see reason at https://github.com/checkstyle/checkstyle/pull/12842#discussion_r1142077252

Reply:
Leaving this unresolved so that others can see testing. Let's do testing and share in PR description for next one.

Reply:
![image](https://user-images.githubusercontent.com/90137881/226462256-e1697488-55ee-4476-a146-946983bd07bd.png)
Sharing this in depth explanation of what happened for the benefit of new contributor.